### PR TITLE
cmd/syncthing/cli: Add showing pending folders for given device (fixes #8130)

### LINKED ISSUE #8130: CLI: add command to print pending folders offered by specific device
The syncthing CLI allows to show pending devices and folders with the commands:
```
syncthing cli show pending devices
syncthing cli show pending folders
```
respectively. From the API [GET /rest/cluster/pending/folders](https://docs.syncthing.net/rest/cluster-pending-folders-get.html):

> Lists folders which remote devices have offered to us, but are not yet shared from our instance to them. Takes the optional device parameter to only return folders offered by a specific remote device. Other offering devices are also omitted from the result.

Therefore I think it would be good to sync the CLI with the API and add a command that takes a deviceId parameter and prints pending folders offered by that specific device.

### PR DESCRIPTION
**Pull request by DanielBarczyk**
Added command
`syncthing cli show pending device-folders <device-id>`
which is a wrapper for [this API call](https://docs.syncthing.net/rest/cluster-pending-folders-get.html).

### Purpose

fixes #8130

### Testing

All tests pass.

### CODE REVIEW

**Comment by imsodin** [comment]
I'd add it as argument, just like in the API: `syncthing cli show pending folders --device <device-id>` or maybe `--offered-by`. It's the same command/information, just filtered.

File: cmd/syncthing/cli/pending.go [hunk 1]
```diff
@@ -25,5 +25,16 @@ var pendingCommand = cli.Command{
...
+		{
+			Name:      "device-folders",
+			Usage:     "Show pending folders for a given device",
+			ArgsUsage: "[device id]",
+			Action:    expects(1, device_folders),
+		},
 	},
 }
+
+func device_folders(c *cli.Context) error {
```
Comment (AudriusButkevicius) [inline cmd/syncthing/cli/pending.go:37]: No snake case in go pls

File: cmd/syncthing/cli/utils.go [hunk 2]
```diff
@@ -58,6 +58,23 @@ func indexDumpOutput(url string) cli.ActionFunc {
...
 
+func indexDumpOutputWithQuery(url string, query map[string]string) cli.ActionFunc {
+	return func(c *cli.Context) error {
+		client, err := getClientFactory(c).getClient()
+		if err != nil {
+			return err
+		}
+		response, err := client.GetWithQuery(url, query)
+		if errors.Is(err, errNotFound) {
+			return errors.New("not found (folder/file not in database)")
```
Comment (AudriusButkevicius) [inline cmd/syncthing/cli/utils.go:69]: Seems wrong message?

**Comment by AudriusButkevicius** [comment]
I might be tripping, but isn't this available as part of the config already?

**Comment by acolomb** [comment]
> I might be tripping, but isn't this available as part of the config already?

No, they were moved out of the config.

I agree with @imsodin that an option to the existing command would be the better approach.

**Comment by DanielBarczyk** [comment]
I applied @imsodin's suggestion and replaced
`syncthing cli show pending device-folders <device-id>`
with
`syncthing cli show pending folders --device <device-id>`

File: cmd/syncthing/cli/pending.go [hunk 3]
```diff
@@ -21,9 +23,23 @@ var pendingCommand = cli.Command{
 			Action: expects(0, indexDumpOutput("cluster/pending/devices")),
 		},
 		{
-			Name:   "folders",
-			Usage:  "Show pending folders",
-			Action: expects(0, indexDumpOutput("cluster/pending/folders")),
+			Name:  "folders",
+			Usage: "Show pending folders",
+			Flags: []cli.Flag{
+				cli.StringFlag{Name: "device", Usage: "Show pending folders for given device"},
```
Comment (imsodin) [inline cmd/syncthing/cli/pending.go:29]:
```suggestion
				cli.StringFlag{Name: "device", Usage: "Show pending folders offered by given device"},
```

**Review by imsodin — DISMISSED** [review]
Looks good, thanks for adjusting. One text comment that I don't feel very strongly about, thus already approved.

**Review by acolomb — APPROVED** [review]

**Pull request merged**